torchtune/utils/metric_logging.py

```diff
@@ -183,17 +183,25 @@ def __init__(
 
         _, self.rank = get_world_size_and_rank()
 
-        if self.rank == 0:
+        if self._wandb.run is None and self.rank == 0:
+            # we check if wandb.init got called externally,
             run = self._wandb.init(
                 project=project,
                 entity=entity,
                 group=group,
-                reinit=True,
-                resume="allow",
                 dir=self.log_dir,
                 **kwargs,
             )
-            run._label(repo="torchtune")
+
+        if self._wandb.run:
+            self._wandb.run._label(repo="torchtune")
```

Comment:
nit: is this different from just keeping the `run._label(repo="torchtune")` line in the if statement above?